File: Python/crosscountry/CrossCountry.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Scanner;

public class CrossCountry {
   public class Info implements Comparable<Info> {
      int loc;
      ArrayList<Integer> vis;
      int min;

      public Info(int location, ArrayList<Integer> visited, int minutes) {
         loc = location;
         vis = visited;
         min = minutes;
      }

      public int compareTo(Info that) {
         return this.min - that.min;
      }
   }

   public static void main(String[] args) {
      Scanner in = new Scanner(System.in);

      String[] nst = in.nextLine().split(" ");

      int n = Integer.parseInt(nst[0]);
      int car = Integer.parseInt(nst[1]);
      int dest = Integer.parseInt(nst[2]);
      int[][] grid = new int[n][n];
      LinkedList<Info> queue = new LinkedList<Info>();

      for (int x = 0; x < n; x++) {
         String[] line = in.nextLine().split(" ");
         for (int y = 0; y < n; y++) {
            grid[x][y] = Integer.parseInt(line[y]);
         }
      }
      in.close();

      queue.push(new Info(car, new ArrayList<Integer>(), 0));
   }  
}

File: Python/crosscountry/CrossCountry.py
from queue import Queue

#Class that stores: current location, visited, and total minutes
class Info():
   def __init__(self, loc, vis, mins):
      self.loc = loc
      self.vis = vis
      self.mins = mins

nst = input().split()

n = int(nst[0])
car = int(nst[1])
dest = int(nst[2])
grid = list()
minutes = float("inf")
queue = Queue()

for x in range(n):
   line = input().split()
   grid.append(line)
   
queue.put(Info(car, set(), 0))

while not queue.empty():
   curr = queue.get()
   
   if curr.loc == dest:
      if curr.mins < minutes:
         minutes = curr.mins
      continue
   
   visited = curr.vis
   visited.add(curr.loc)
      
   for i in range(n):
      if i in visited:
         continue
      
      location = int(grid[curr.loc][i])
      if curr.mins + location > minutes:
         continue
      queue.put(Info(i, visited, curr.mins + location))
      
print(int(minutes))